lib/elixir/lib/protocol.ex

```diff
@@ -806,6 +806,16 @@ defmodule Protocol do
         nil
       )
     end
+
+    quote do
+      # Register the __impl__/1 callback to ensure the protocol is a behaviour
+      # Without this, a protocol that defines no functions won't be detected
+      # as a behaviour by the Erlang compiler, and any implementations of that
+      # protocol will log a warning.
+      @callback __impl__(:foo) :: module()
```

Comment:
Is `:foo` correct here? :)

Reply:
It absolutely is not! That's intended to say `:for` 😅 I've just put my laptop away for the night, but I can fix it up tomorrow morning

Reply:
No worries, take your time!

```suggestion
      @callback __impl__(:for) :: module()
```